packages/bridges/src/SignatureBridgeSide.ts

```diff
@@ -18,60 +18,52 @@ export class SignatureBridgeSide implements IBridgeSide {
-    signer: ethers.Signer,
-    signingSystemSignFn?: (data: any) => Promise<string>
+    systemSigningFn: SystemSigningFn,
   ) {
     this.contract = contract;
-    this.admin = signer;
-    this.governor = governor;
     this.anchorHandler = null;
     this.tokenHandler = null;
     this.treasuryHandler = null;
     this.proposals = [];
-    if (signingSystemSignFn) {
-      // The signing system here is an asynchronous function that
-      // potentially dispatches a message for a signature and waits
-      // to receive it. It is potentially a long-running process.
-      this.signingSystemSignFn = signingSystemSignFn;
-    } else {
-      if (typeof governor === 'string') {
-        throw new Error('Cannot sign with signing system without a governor wallet');
-      }
-
-      this.signingSystemSignFn = (data: any) => {
-        return Promise.resolve(signMessage(governor, data));
-      };
-    }
+
+    this.signingSystemSignFn = systemSigningFn
   }
 
   public static async createBridgeSide(
-    initialGovernor: ethers.Wallet | string,
-    admin: ethers.Signer,
-    signingSystemSignFn?: (data: any) => Promise<string>
+    admin: ethers.Wallet
   ): Promise<SignatureBridgeSide> {
     const bridgeFactory = new SignatureBridge__factory(admin);
-    const deployedBridge = (typeof initialGovernor === 'string')
-      ? await bridgeFactory.deploy(initialGovernor, 0)
-      : await bridgeFactory.deploy(initialGovernor.address, 0);
+    const deployedBridge = await bridgeFactory.deploy(admin.address, 0);
     await deployedBridge.deployed();
-    const bridgeSide = (typeof initialGovernor === 'string')
-      ? new SignatureBridgeSide(deployedBridge, initialGovernor, admin, signingSystemSignFn)
-      : new SignatureBridgeSide(deployedBridge, initialGovernor, admin, signingSystemSignFn);
+    const bridgeSide = new SignatureBridgeSide(deployedBridge, (data: string) => {
+      return Promise.resolve(signMessage(admin,data));
+    });
     return bridgeSide;
   }
 
-  public static async connect(address: string, governor: ethers.Wallet, admin: ethers.Wallet) {
-    const deployedBridge = SignatureBridge__factory.connect(address, admin);
-    const bridgeSide = new SignatureBridgeSide(deployedBridge, governor, admin);
+  public static async connectMocked(contractAddress: string, mockedGovernor: ethers.Wallet) {
+    const deployedBridge = SignatureBridge__factory.connect(contractAddress, mockedGovernor);
+    const bridgeSide = new SignatureBridgeSide(deployedBridge, (data: string) => {
+      return Promise.resolve(signMessage(mockedGovernor,data));
+    });
+    return bridgeSide;
+  }
+
+  public static async connectGovernor(
+    contractAddress: string,
+    provider: ethers.providers.Provider,
+    systemSigningFn: SystemSigningFn
+  ) {
+    const deployedBridge = SignatureBridge__factory.connect(contractAddress, provider);
+    const bridgeSide = new SignatureBridgeSide(deployedBridge, systemSigningFn);
     return bridgeSide;
   }
```

Comment:
A few comments here would make things easier to understand. Like what are these two method does, and what is the difference? a use case(s) in the comments would make much sense to understand the difference between both.

Reply:
Added some comments explaining the use of both connectMocked and connectGovernor.